subsys/bluetooth/host/adv.c

```diff
@@ -49,13 +49,23 @@ struct ad_stream {
 	 * The length and type are included in this offset.
 	 */
 	uint16_t current_ltv_offset;
+
+	/* The remaining size of total ad[i].data[j].data_len + 2 for LTV header */
+	uint16_t remaining_size;
 };
 
 static void ad_stream_new(struct ad_stream *stream, const struct bt_ad *ad, size_t ad_len)
 {
 	(void)memset(stream, 0, sizeof(*stream));
 	stream->ad = ad;
 	stream->ad_len = ad_len;
+
+	for (int i = 0; i < ad_len; i++) {
+		for (int j = 0; j < ad[i].len; j++) {
+			/* LTV length + type + value */
+			stream->remaining_size += ad[i].data[j].data_len + 2;
```

Comment:
remaining_size `uint16_t` max 65535, how can overflow?

Reply:
since ad_len is a `size_t`, the application can technically attempt to set 22000 entries in the array, and if they are each 3 octets (length + type + 1 byte data) that will be a total of 66000 octets here, and that would cause an overflow, and `stream->remaining_size` will be `465`.

Reply:
Done

Reply:
What is "done" here? I don't see any change

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/43255/files#r822427197

Reply:
I'm sorry, I don't fully understand your comment. 

My point is that you are here, without conditions, increasing a `size_t` value by looping over two other `size_t` values. Without any checks for `ad_len` or `ad[i].len` this can cause overflow.

Reply:
```suggestion
			please provide your modification
```

Reply:
Now that the type of remaining_size  is already size_t, how can this possibly overflow?

Reply:
According to your opinion, `current_ltv_offset` this parameter may still overflow, isn't it?

Reply:
If four bytes of space cannot represent the length of the total data, the use of the application layer is undoubtedly stupid.

Reply:
> If four bytes of space cannot represent the length of the total data, the use of the application layer is undoubtedly stupid.

Right, and we should protect the stack against a buggy (or stupid) application. 

We know that the maximum advertising data can be 1650. 

I would suggest simply adding
```c
if (stream->remaining_size > BT_GAP_ADV_MAX_EXT_ADV_DATA_LEN) {
	/* return err */
}
```

Which not only guards against overflow, but also attempted spec violation.